modules/Movecraft/src/main/java/net/countercraft/movecraft/commands/SiegeCommand.java

```diff
@@ -297,4 +294,29 @@ private void displayInfo(@NotNull CommandSender sender, @NotNull Siege siege) {
         sender.sendMessage(I18nSupport.getInternationalisedString("Siege - End Time") + end + militaryTimeIntToString(siege.getScheduleEnd()) + " UTC");
         sender.sendMessage(I18nSupport.getInternationalisedString("Siege - Duration") + ChatColor.WHITE + secondsIntToString(siege.getDuration()));
     }
+
+    @Override
+    public List<String> onTabComplete(CommandSender commandSender, Command command, String s, String[] strings) {
+        final List<String> tabCompletions = new ArrayList<>();
+        if (strings.length <= 1) {
+            tabCompletions.add("info");
+            tabCompletions.add("begin");
+            tabCompletions.add("list");
+        } else if (strings[0].equalsIgnoreCase("info")) {
+            for (Siege siege : Movecraft.getInstance().getSiegeManager().getSieges()) {
+                tabCompletions.add(siege.getName());
+            }
+        }
+        if (strings.length == 0) {
+            return tabCompletions;
+        }
+        final List<String> completions = new ArrayList<>();
+        for (String completion : tabCompletions) {
+            if (!completion.startsWith(strings[strings.length - 1])) {
+                continue;
+            }
+            completions.add(completion);
```

Comment:
Why not invert the logic of the if and put ` completions.add(completion);` inside?

Reply:
Gives a lot tidier code by avoiding too many "nestings"